src/test/java/dk/kb/present/solr/EmbeddedSolrTest.java

```diff
@@ -449,6 +450,47 @@ void testTitle() throws Exception {
 		assertEquals("Romeo og Julie", record.getFieldValue("title"));
 	}
 
+	@Test
+	void testImageResource() throws Exception {
+		String solrString = TestUtil.getTransformedWithAccessFieldsAdded(MODS2SOLR, RECORD_40221e30);
+
+		Gson gson = new GsonBuilder().setPrettyPrinting().create();
+		JsonElement je = JsonParser.parseString(solrString);
+		String prettyJsonString = gson.toJson(je);
+		//System.out.println(prettyJsonString);
+
+		SolrInputDocument document = convertJsonToSolrJavaDoc(prettyJsonString);
+		embeddedServer.add(document);
+		embeddedServer.commit();
+		assertEquals(1, getNumberOfTotalDocuments());
+
+		//Full life cycle test
+		SolrDocument record = getRecordById("40221e30-1414-11e9-8fb8-00505688346e");
+
+		//Single value field
+		assertEquals("/DAMJP2/DAM/Samlingsbilleder/0000/624/420/KE070592", record.getFieldValue("image_resource"));
+	}
+
+	@Test
+	void testMapScale() throws Exception {
+		String solrString = TestUtil.getTransformedWithAccessFieldsAdded(MODS2SOLR, RECORD_0c02aa10);
+
+		Gson gson = new GsonBuilder().setPrettyPrinting().create();
+		JsonElement je = JsonParser.parseString(solrString);
+		String prettyJsonString = gson.toJson(je);
+
+		SolrInputDocument document = convertJsonToSolrJavaDoc(prettyJsonString);
+		embeddedServer.add(document);
+		embeddedServer.commit();
+		assertEquals(1, getNumberOfTotalDocuments());
+
+		//Full life cycle test
+		SolrDocument record = getRecordById("0c02aa10-b657-11e6-aedf-00505688346e");
+
+		//Single value field
+		assertEquals("Målestok 1:75 000",record.getFieldValue("map_scale"));
```

Comment:
Lots of duplicated code here. What about a helper method?

Reply:
Ofcourse it should. So many lines of codes now deleted!